src/LtiAssignmentsGradesService.php

```diff
@@ -104,11 +80,8 @@ public function findOrCreateLineitem(LtiLineitem $newLineItem): LtiLineitem
 
     public function getGrades(LtiLineitem $lineitem = null)
     {
-        if ($lineitem !== null) {
-            // Skip line item lookup/creation if we already know the Id
-            if (empty($lineitem->getId())) {
-                $lineitem = $this->findOrCreateLineitem($lineitem);
-            }
+        if (isset($lineitem)) {
+            $lineitem = $this->ensureLineItemExists($lineitem);
```

Comment:
I think you don't need to check if $lineitem isset here to use ensureLineItemExists. If $lineitem is null, then ensureLineItemExists will try to pull the lineitem from the service data (using getResourceLaunchLineItem) and if that fails, will create a default line item much like putGrade() does if you call it will a null line item. thoughts?

Reply:
but if you just wanted to avoid changing behavior of getGrades, that is also fine.

Reply:
I went back and forth on this, trying to decide what makes most sense if you try to call `getGrades()` when no results URL exists. I was torn between throwing an exception (my previous code) and just creating a default line item (your suggestion). I don't think the LTI spec specifies what should happen, and creating the default line item gives cleaner code, so I'm going with your suggestion.

Reply:
Sounds good -- I'm also fine with throwing an exception. I just think it should do the same in putGrade() if we decide to do it in getGrades.